metric_providers/base.py

```diff
@@ -47,7 +47,17 @@ def __init__(
 
     # this is the default function that will be overridden in the children
     def check_system(self):
-        pass
+        cmd = ['pgrep', '-f', self._metric_provider_executable]
+        result = subprocess.run(cmd,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            check=False, encoding='UTF-8')
+        # returncode 1 sadly indicates program failure or no found process, so we need to go deeper
```

Comment:
The comment here could be a little clearer. 
```
EXIT STATUS
     The pgrep and pkill utilities return one of the following values upon exit:

     0       One or more processes were matched.

     1       No processes were matched.

     2       Invalid options were specified on the command line.

     3       An internal error occurred.
```
The code looks ok though

Reply:
I restructured it to make it clearer when reading the code